src/source.cc

```diff
@@ -2120,7 +2121,25 @@ bool Source::View::on_button_press_event(GdkEventButton *event) {
     get_buffer()->select_range(start, end);
     return true;
   }
-  
+  if((event->type == GDK_BUTTON_PRESS) && (event->button == 3)){
+    auto menu=Menu::get().edit_menu;
```

Comment:
I was thinking that cut/copy for instance is not needed in the menu when you right click on a buffer without selected text. And Rename for instance is not needed when you right click on a buffer with selected text.